src/wp-admin/includes/class-wp-filesystem-direct.php

```diff
@@ -331,12 +338,18 @@ public function move( $source, $destination, $overwrite = false ) {
 			return false;
 		}
 
+		if ( $overwrite && $this->exists( $destination ) && ! $this->delete( $destination, true ) ) {
+			// Can't overwrite if the destination couldn't be deleted.
+			return false;
+		}
+
 		// Try using rename first. if that fails (for example, source is read only) try copy.
 		if ( @rename( $source, $destination ) ) {
 			return true;
 		}
 
-		if ( $this->copy( $source, $destination, $overwrite ) && $this->exists( $destination ) ) {
+		// Backwards Compatibility: Only fall back to `::copy()` for single files.
+		if ( $this->is_file( $source ) && $this->copy( $source, $destination, $overwrite ) && $this->exists( $destination ) ) {
```

Comment:
Unsure what is best here. 🤔

It's inconsistent to fallback for files but not directories. The aim of this PR is to improve consistency. 

Are you able to explain your thinking?

Reply:
Yep, the issue is that removing the original conditions would introduce a backwards compatibility break, and for `WP_Filesystem_Direct`, which is used by 95%+ of sites. To maintain previous behaviour without it erroring when a directory is passed, `$this->is_file( $source )` protects BC.

Unfortunately, an inconsistency that we may be stuck with.